app/api/users/route.ts

```diff
@@ -0,0 +1,67 @@
+import pool from '@/lib/db';
+
+export async function GET(request: Request) {
+    try {
+
+        const url = new URL(request.url);
+        const query = url.searchParams.get('query') || '';
+        const limit = Number(url.searchParams.get('limit') || 10);
+        const page = Number(url.searchParams.get('page') || 1);
+        const type = url.searchParams.get('type') || 'recommended';
+        const topics = url.searchParams.getAll('topics[]');
+        const author_id = url.searchParams.get('author_id') || null;
+        const offset = (page - 1) * limit;
+
+        let queryText = `
+            SELECT 
+                id, 
+                name, 
+                image, 
+                bio, 
+                link, 
+                follower_count, 
+                following_count 
+            FROM users
+        `;
+        const queryParams: any[] = [];
+        const whereClauses = [];
+
+        if (query) {
+            queryParams.push(`%${query}%`);
+            whereClauses.push(`(name ILIKE $${queryParams.length} OR bio ILIKE $${queryParams.length})`);
+        }
+
+        if (topics.length > 0) {
+            queryParams.push(topics);
+            whereClauses.push(`id IN (SELECT user_id FROM users_topics WHERE topic_name = ANY($${queryParams.length}))`);
+        }
+
+        if (type === 'recommended') {
+            queryText += ` ORDER BY follower_count DESC, following_count DESC`;
+        } else if (type === 'followers' && author_id) {
+            queryParams.push(author_id);
+            whereClauses.push(`id IN (SELECT follower_id FROM follows WHERE following_id = $${queryParams.length})`);
+        } else if (type === 'following' && author_id) {
+            queryParams.push(author_id);
+            whereClauses.push(`id IN (SELECT following_id FROM follows WHERE follower_id = $${queryParams.length})`);
+        } else if (type === 'followers' || type === 'following') {
+            return new Response(JSON.stringify({ error: 'author_id is required for followers/following type' }), { status: 400 });
+        }
+
+        if (whereClauses.length > 0) {
+            queryText += ' WHERE ' + whereClauses.join(' AND ');
+        }
```

Comment:
Where clause should come before the Order by clause, maybe you can check if type===recommended after you've appended all where clauses.